dasharo-compatibility/check-ethernet-ports-order.robot

```diff
@@ -0,0 +1,69 @@
+*** Settings ***
+Library             Collections
+Library             DateTime
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=20 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+
+Suite Setup         Run Keywords
+...                     Prepare Test Suite
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+SPS001.001 Ethernet ports are in order
+    [Documentation]    This test automates the verification of port order based
+    ...    on PCIe bus numbers and checks PCIe switching.
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${pci_devices}=    Get MACs
+    Log    PCIe devices and their MACs: ${pci_devices}
+    Compare Interfaces    ${pci_devices}
+
+
+*** Keywords ***
+Get PCIe
+    ${lspci_output}=    Execute Command In Terminal    lspci -nn |grep "${ETHERNET_ID}"
+    @{lines}=    Split String    ${lspci_output}    \n
+    ${ethernet_devs}=    Create List
+    FOR    ${line}    IN    @{lines}
+        ${dev}=    Evaluate    '${line}'.split()[0]
+        Append To List    ${ethernet_devs}    ${dev}
+    END
+    RETURN    ${ethernet_devs}
+
+Get MACs
+    [Documentation]    Extract MAC addresses from lspci output.
+    ${ethernet_devs}=    Get PCIe
+    ${pci_devices}=    Create List
+    FOR    ${dev}    IN    @{ethernet_devs}
+        ${lspci_output}=    Execute Command In Terminal    lspci -s ${dev} -v |grep "Device Serial"
+        Append PCIe MAC    ${lspci_output}    ${pci_devices}
+    END
+    RETURN    ${pci_devices}
+
+Append PCIe MAC
+    [Arguments]    ${line}    ${pci_devices}
+    ${serial}=    Evaluate    '${line}'.split()[-1]
+    ${mac}=    Replace String    ${serial}    ff-ff-    ${EMPTY}
+    IF    '00-00-00-00-00-00' not in '${mac}'
+        Append To List    ${pci_devices}    ${mac}
+    END
+
+Compare Interfaces
+    [Arguments]    ${pci_devices}
+    Log    Sorted interfaces: ${ETH_PORTS} \n
+    Log    Sorted PCIe devices: ${pci_devices} \n
```

Comment:
```suggestion
```

No need to log it. The lists will be visible anyways in the `Should Be Equal As Strings` keyword in the log.

Reply:
ok, https://github.com/Dasharo/open-source-firmware-validation/pull/515/commits/35fd5151cb4fae97bcf4f53aaacdbdea03a74325